frontend/src/components/StockData/StockData.jsx

```diff
@@ -74,22 +76,28 @@ export default function StockData(props) {
     return (
         <div className="stock-data">
             {isLoading ? (
-                <p>Loading...</p>
+                <div className="loading">
+                    <div className="ball"></div>
+                    <p>Loading...</p>
+                </div>
             ) : stockNotFound ? (
-                <p>Stock Not Found...</p>
+                <div className="stock-not-found">
+                    <p>Stock Not Found...</p>
+                </div>
             ) : (
```

Comment:
Here you have three nested conditions, That is not recommended with ternaries, nor with IF-ELSE clauses.
Here's a better alternative:

```
{isLoading ? (
  <jsx />
) : null}

{stockNotFound ? (
  <jsx />
) : (
  <jsx />
)}
```